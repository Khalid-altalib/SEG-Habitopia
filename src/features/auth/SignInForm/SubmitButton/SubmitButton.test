// // Import necessary packages and components
// import React, { BaseSyntheticEvent } from "react";
// import { fireEvent, render, waitFor } from "@testing-library/react-native";
// import { Provider } from "react-redux";
// import configureStore from "redux-mock-store";
// import SubmitButton from "../SubmitButton";
// import { SignInFormValues } from "../../../../../types";
// import { addLogInData, logInUser } from "../../authSlice";

// // Create a mock store with the desired state
// const mockStore = configureStore([]);
// const store = mockStore({
//   // Add any necessary initial state here
// });

// const mockHandleSubmit = jest
//   .fn()
//   .mockImplementation((onSubmit: (data: SignInFormValues) => Promise<void>) => {
//     return async (event?: BaseSyntheticEvent<object, any, any>) => {
//       await onSubmit({
//         email: "test@example.com",
//         password: "testpassword",
//       });
//     };
//   });

// describe("SubmitButton", () => {
//   test("calls addLogInData and logInUser actions when submitted", async () => {
//     const dispatch = jest.fn();
//     const wrapper = render(<SubmitButton handleSubmit={mockHandleSubmit} />, {
//       wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
//     });

//     await fireEvent.press(wrapper.getByText("Continue"));

//     expect(dispatch).toHaveBeenCalledWith(
//       addLogInData({ email: "test@example.com", password: "testpassword" })
//     );
//     expect(dispatch).toHaveBeenCalledWith(logInUser());
//   });
//   it("should call the onSubmit function when the button is pressed", async () => {
//     const mockHandleSubmit = jest.fn().mockImplementation((onSubmit) => {
//       return async () => {
//         await onSubmit({ email: "test@example.com", password: "password" });
//       };
//     });
//     const { getByText } = render(
//       <SubmitButton handleSubmit={mockHandleSubmit} />
//     );
//     const button = getByText("Continue");
//     fireEvent.press(button);
//     await waitFor(() => expect(mockHandleSubmit).toHaveBeenCalledTimes(1));
//     expect(mockHandleSubmit.mock.calls[0][0]).toBeInstanceOf(Function);
//   });
// });
