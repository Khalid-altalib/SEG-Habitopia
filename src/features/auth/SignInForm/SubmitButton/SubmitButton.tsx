// React Hook Form
import { UseFormHandleSubmit } from "react-hook-form";

// Expo Vector Icons
import { AntDesign } from "@expo/vector-icons";

// Habitopia
import Button from "../../../../components/Button";
import { addLogInData, logInUser } from "../../authSlice";
import { useAppDispatch } from "../../../../app/hooks";
import { ButtonType, SignInFormValues } from "../../../../../types";

/**
 * A button for submitting the sign in form.
 *
 * @param props The properties passed to the component.
 * @returns The component representing the submit button.
 */

const SubmitButton = ({ handleSubmit }: Props) => {
  const dispatch = useAppDispatch();

  const onSubmit = async (data: SignInFormValues) => {
    dispatch(addLogInData(data));

    await dispatch(logInUser());
  };

  return (
    <Button
      style={{ marginTop: 20 }}
      isFullWidth
      icon={<AntDesign name="arrowright" size={20} color="white" />}
      type={ButtonType.Primary}
      onPress={handleSubmit(onSubmit)}
    >
      Continue
    </Button>
  );
};

type Props = {
  /** A callback used by React Hook Form to handle submit, i.e. validate the form and act as an error callback. */
  handleSubmit: UseFormHandleSubmit<SignInFormValues>;
};

export default SubmitButton;
