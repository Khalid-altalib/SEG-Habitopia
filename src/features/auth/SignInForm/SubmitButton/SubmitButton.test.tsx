import React from "react";
import { render } from "@testing-library/react-native";
import SubmitButton from "./SubmitButton";
import TestingWrapper from "@app/testingWrapper";
import configureStore from "redux-mock-store";
import { useSelector } from "react-redux";

const mockStore = configureStore([]);

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
}));

describe("SubmitButton", () => {
  it("renders correctly", () => {
    (useSelector as jest.Mock).mockReturnValue({
      loading: false,
    });

    const store = mockStore({});
    const handleSubmit = jest.fn();

    const wrapper = render(
      <TestingWrapper store={store}>
        <SubmitButton handleSubmit={handleSubmit} />
      </TestingWrapper>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
