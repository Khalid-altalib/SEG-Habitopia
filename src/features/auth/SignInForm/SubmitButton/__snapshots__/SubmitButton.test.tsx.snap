// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    icon={
      <Icon
        allowFontScaling={false}
        color="white"
        name="arrowright"
        size={20}
      />
    }
    isFullWidth={true}
    isLoading={false}
    style={
      {
        "marginTop": 20,
      }
    }
    type={0}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 20,
          "opacity": 1,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          {
            "colors": [
              4278220774,
              4286456526,
            ],
            "endPoint": [
              1,
              1,
            ],
            "locations": undefined,
            "startPoint": [
              0,
              0,
            ],
          }
        }
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 10,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 40,
              "paddingVertical": 15,
            },
            {
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#18181b",
              },
              {
                "color": "#e4e4e7",
                "fontFamily": undefined,
                "fontSize": 19,
              },
              [
                null,
                {
                  "alignSelf": "center",
                },
              ],
            ]
          }
        >
          Continue
        </Text>
        <View
          style={
            {
              "marginLeft": 5,
            }
          }
        >
          <Text />
        </View>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
